tests/integration/test_spark_operator_integration.py

```diff
@@ -0,0 +1,30 @@
+import logging
+import pytest
+from tenacity import retry, wait_exponential, stop_after_delay, before_log
+from lightkube.resources.core_v1 import Pod, ServiceAccount
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test, helpers):
+    spark_operator_charm = await ops_test.build_charm(".")
+
+    spark_resources = {"oci-image": helpers.oci_image("./metadata.yaml", "oci-image")}
+    spark_app_name = "spark"
+    await ops_test.model.deploy(spark_operator_charm, resources=spark_resources, application_name=spark_app_name)
+    await ops_test.model.wait_for_idle(status="active")
```

Comment:
Generally, charms entering blocked status and resolving themselves is an error we want to catch (blocked should mean "needs manual intervention").  Good to throw this in here unless there's a reason not to.  And the timeout helps fail faster (not sure if 5 min is appropriate for this or not, but you probably have a sense of what a reasonable timeline for deployment should be)

```suggestion
    await ops_test.model.wait_for_idle(status="active", raise_on_blocked=True, timeout=300)
```

Reply:
added in most recent commit